clients/gtest/her2k_gtest.cpp

```diff
@@ -0,0 +1,163 @@
+/* ************************************************************************
+ * Copyright 2020 Advanced Micro Devices, Inc.
+ *
+ * ************************************************************************ */
+
+#include "rocblas_data.hpp"
+#include "rocblas_datatype2string.hpp"
+#include "rocblas_test.hpp"
+#include "testing_her2k.hpp"
+#include "testing_her2k_batched.hpp"
+#include "testing_her2k_strided_batched.hpp"
+#include "type_dispatch.hpp"
+#include <cstring>
+#include <type_traits>
+
+namespace
+{
+    // possible gemv test cases
+    enum her2k_test_type
+    {
+        HER2K,
+        HER2K_BATCHED,
+        HER2K_STRIDED_BATCHED,
+    };
+
+    //ger test template
+    template <template <typename...> class FILTER, her2k_test_type HER2K_TYPE>
+    struct her2k_template : RocBLAS_Test<her2k_template<FILTER, HER2K_TYPE>, FILTER>
+    {
+        // Filter for which types apply to this suite
+        static bool type_filter(const Arguments& arg)
+        {
+            return rocblas_simple_dispatch<her2k_template::template type_filter_functor>(arg);
+        }
+
+        // Filter for which functions apply to this suite
+        static bool function_filter(const Arguments& arg)
+        {
+            switch(HER2K_TYPE)
+            {
+            case HER2K:
+                return !strcmp(arg.function, "her2k") || !strcmp(arg.function, "her2k_bad_arg");
+            case HER2K_BATCHED:
+                return !strcmp(arg.function, "her2k_batched")
+                       || !strcmp(arg.function, "her2k_batched_bad_arg");
+            case HER2K_STRIDED_BATCHED:
+                return !strcmp(arg.function, "her2k_strided_batched")
+                       || !strcmp(arg.function, "her2k_strided_batched_bad_arg");
+            }
+            return false;
+        }
+
+        // Google Test name suffix based on parameters
+        static std::string name_suffix(const Arguments& arg)
+        {
+            RocBLAS_TestName<her2k_template> name;
+
+            name << rocblas_datatype2string(arg.a_type);
+
+            if(strstr(arg.function, "_bad_arg") != nullptr)
+            {
+                name << "_bad_arg";
+            }
+            else
+            {
+                // alpha & beta only real
```

Comment:
Only beta is real in her2k right? Lines 87-89 can be removed, but it's insignificant.